docs/TEST_GUIDE.md

```diff
@@ -0,0 +1,66 @@
+# Test guide
+
+This project uses 3 layers of automated tests (unit, integration, and end-to-end). 
+It is very powerful setup, but at the same time it can be challenging to choose appropriate 
+level of test for the problem at hand. 
+This is the reason why this guide was created. 
+
+The goal of this guide is to provide an overview of used types of tests, describe their pros and cons, and give hints for when to use which.
+
+## Unit tests
+
+### What do they test?
+One unit ([component](/LiskHQ/lisk-hub/blob/development/src/components),
+[util](/LiskHQ/lisk-hub/blob/development/src/utils),
+[action](/LiskHQ/lisk-hub/blob/development/src/actions),
+[reducer](/LiskHQ/lisk-hub/blob/development/src/store/reducers),
+[middleware](/LiskHQ/lisk-hub/blob/development/src/store/middlewares), or
+[subscriber](/LiskHQ/lisk-hub/blob/development/src/store/subscribers)) in isolation.
+
+### How are they organized?
+Each unit test live in the same folder as the unit that it tests. E.g. tests for [/src/components/login/login.js](/LiskHQ/lisk-hub/blob/development/src/components/login/login.js) are in [/src/components/login/login.test.js](/LiskHQ/lisk-hub/blob/development/src/components/login/login.test.js).
+
+Configuration is in [karma.conf.js](/LiskHQ/lisk-hub/blob/development/karma.conf.js) and [config/webpack.config.test.js](/LiskHQ/lisk-hub/blob/development/config/webpack.config.test.js).
+
+### What tools are used?
+- **Assertions** use `expect` syntax of [chai](http://www.chaijs.com/), e.g.:
+https://github.com/LiskHQ/lisk-hub/blob/80afda8289b75cb70bf345d14d36117fde2bdd90/src/utils/passphrase.test.js#L136
+
+- **Spies, stubs, ands mocks** use [sinon](http://sinonjs.org/), e.g.:
+https://github.com/LiskHQ/lisk-hub/blob/19a2a613fa08a34fe3088b0e40c11c7fa37a645d/src/store/middlewares/savedSettings.test.js#L20
+
+- To test **React compoments** we use [enzyme](http://airbnb.io/enzyme/), e.g.:
```

Comment:
Should we mention here the helper `chai-enzyme` that can be used also to simplify tests?

Reply:
Overall, I miss some brief/general hints on exclusion of tests between the different layers. 

Scenario: sumComponent - a components that sums two numbers. 
- unit --> test that the sum is correct
- integration 
-- test that the sum is displayed, whatever it is, does not need to assert that is actually correct. 
-- test that when the sum is displayed, no other components are affected
- e2e
-- we can assert here that the sum contains the actual production value expected, the rest is already covered?

Reply:
Stateless components will need unit tests only if they're not fully covered by their stateful parents.